src/main/java/org/gridsuite/gateway/filters/TokenValidatorGlobalPreFilter.java

```diff
@@ -109,29 +121,68 @@ public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
         ClientID clientID = new ClientID(jwtClaimsSet.getAudience().get(0));
 
         JWSAlgorithm jwsAlg = JWSAlgorithm.parse(jwt.getHeader().getAlgorithm().getName());
-        return gatewayService.getJwksUrl(jwtClaimsSet.getIssuer())
-                .flatMap(jwkSet -> {
-                    try {
-                        URL jwkSetURL = new URL(jwkSet);
-
-                        // Create validator for signed ID tokens
-                        IDTokenValidator validator = new IDTokenValidator(iss, clientID, jwsAlg, jwkSetURL);
-
-                        validator.validate(jwt, null);
-                        // we can safely trust the JWT
-                        LOGGER.debug("Token verified, it can be trusted");
-
-                        //we add the subject header
-                        exchange.getRequest()
-                                .mutate()
-                                .headers(h -> h.set(HEADER_USER_ID, jwtClaimsSet.getSubject()));
-
-                        return chain.filter(exchange);
-                    } catch (JOSEException | BadJOSEException | MalformedURLException e) {
-                        LOGGER.info("{}: 401 Unauthorized, The token cannot be trusted : {}", exchange.getRequest().getPath(), e.getMessage());
-                        return completeWithCode(exchange, HttpStatus.UNAUTHORIZED);
-                    }
+        return jwkUriMap.get(iss.getValue()) != null ? proceedFilter(exchange, chain, jwt, jwtClaimsSet, iss, clientID, jwsAlg, jwkUriMap.get(iss.getValue())) : gatewayService.getJwksUrl(jwtClaimsSet.getIssuer())
```

Comment:
Is really valid since `proceedFilter` does not retrun a `jwkSetUri` ?

Reply:
there is IOException or ParseException catched when invalid uri (function retrieveResource throws IOException) , if invalid we remove it from cache.